code/modules/urist/modules/wfc/modules/deepmaint/admins.dm

```diff
@@ -0,0 +1,42 @@
+// Adminbuse stuff for deepmaint
+
+/datum/admins/proc/generate_deepmaint()
+	set category = "Fun"
+	set desc = "(Re-)Generate DeepmaintRooms; place the map template first!"
+	set name = "Generate DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("failed to generate Deepmaint - no Deepmaint Z-levels registered!")
+		return
+
+	for(var/area/map_template/deepmaint_wfc/deepmaint_area in GLOB.wfc_deepmaint_zlevels_by_instance)
+		var/success = deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json")
+		if(success)
+			log_and_message_admins("has generated a DeepmaintRooms map.")
+			break
+		else
+			log_and_message_admins("has unsuccessfully attempted to generate a DeepmaintRooms map.")
+
+	return
+
+
+/datum/admins/proc/throw_into_deepmaint(var/mob/M)
+	set category = "Fun"
+	set desc = "Throws the target into Deepmaint."
+	set name = "Send to DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("tried to throw [M] into Deepmaint, but no Deepmaint Z-levels are currently registered!")
```

Comment:
Text macro might make sense here, as it's for general mobs?
```suggestion
		log_and_message_admins("tried to throw \a [M] into Deepmaint, but no Deepmaint Z-levels are currently registered!")
```

Reply:
Sure, why not?